magento/pwa-studio
# [PWA-1020] Upgrade Node 14 and Scaffold Node 16

## Description

Upgrade Node version  minimum requirement form Node 10 to Node 14.
Fix issues with NPM 7 and above.

## Related Issue







Closes https://jira.corp.magento.com/browse/PWA-1020.

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature

- Verify scaffold build runs Node 14 (Node 16 is not yet supported on AWS)
- Verify node 14 is by default across all builds

Please locally test on both [Node 14 (Npm 6)](https://nodejs.org/download/release/v14.18.1/) and [Node 16 (Npm 8)](https://nodejs.org/en/).

You can run local scaffolding installation with this command:
`DEBUG_PROJECT_CREATION=1 ./packages/pwa-buildpack/bin/buildpack create-project scaffold-test --template "packages/venia-concept" --name "scaffold-test" --author "Scaffold Test" --backend-url "https://master-7rqtwti-mfwmkrjfqvbjk.us-4.magentosite.cloud/" --npm-client "npm"`

#### Test scenario(s) for any existing impacted features/areas

- Do and overall QA of the website and all critical features.
- Run Storybook.
- Run Cypress tests.

## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
